fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java

```diff
@@ -140,30 +148,45 @@ public String create(final String apiRequestBodyAsJson) {
         return this.toApiJsonSerializer.serialize(result);
     }
 
-    @PUT
-    @Path("{userId}")
-    public String update(@PathParam("userId") final Long userId, final String apiRequestBodyAsJson) {
-
-        final CommandWrapper commandRequest = new CommandWrapperBuilder() //
-                .updateUser(userId) //
-                .withJson(apiRequestBodyAsJson) //
-                .build();
-
-        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-
-        return this.toApiJsonSerializer.serialize(result);
+  @PUT
+  @Path("/{userId}/usrnme/{usrnme}/currentPass/{currentPass}")
```

Comment:
After implementing this code, I did few tests to change the user details. Nothing accept 'updatepassword case' was asked password. For now I have just proposed the change, there are still vulnerabilities in this approach which I discovered later. I am going to Fix them and would come back later. Thanks for the feedback